Doesn't extract passwords from Firefox

I am using both **release 0.2.8** and my build from latest source with this commands:

- .\hack-browser-data.exe -b all -f json -dir results -cc
- .\hack-browser-data.exe -b firefox -f json -dir results -cc
- .\hack-browser-data.exe -b all

It does find Everthing on Edge (version 86.0.622.69) and chrome (version 86.0.4240.198) but no luck in finding passwords from firefox.

**Desktop :**
 - OS Name : Windows 10 2004
 - Browser Name : FireFox
 - Browser Version : 78.0.1 and 82.0.3 (latest)

Additional context :
I have tried both running command in privileged mode and while firefox processes are terminated.
Also logs aren't clear and too much go error in case it doesn't find anything. For firefox and this command `.\hack-browser-data.exe -b all -f json -dir results -cc` logs look like this:
`decrypt_windows.go:122: error asn1: structure error: tags don't match (16 vs {class:0 tag:4 length:20 isCompound:false}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} SequenceC @2
parse.go:523: error decrypt meta data failed asn1: structure error: tags don't match (16 vs {class:0 tag:4 length:20 isCompound:false}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} SequenceC @2
cmd.go:77: error asn1: structure error: tags don't match (16 vs {class:0 tag:4 length:20 isCompound:false}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} SequenceC @2
[x]:  Get 0 passwords, filename is results/firefox_password.json
[x]:  Get 27 bookmarks, filename is results/firefox_bookmark.json
[x]:  Get 854 cookies, filename is results/firefox_cookie.json
[x]:  Get 21277 history, filename is results/firefox_history.json`

--- a/core/data/parse.go
+++ b/core/data/parse.go
@@ -535,11 +535,13 @@ func (p *passwords) FirefoxParse() error {
 		log.Debug("password-check success")
 		m := bytes.Compare(nssA102, keyLin)
 		if m == 0 {
-			nss, err := decrypt.DecodeNss(nssA11)
+			var nss interface{}
+			nss, err = decrypt.DecodeNss(nssA11)
 			if err != nil {
 				log.Error("decode firefox nssA11 bytes failed", err)
 				return err
 			}
+
 			finallyKey, err := decrypt.Nss(globalSalt, masterPwd, nss)
 			finallyKey = finallyKey[:24]
 			if err != nil {

--- a/core/decrypt/decrypt_linux.go
+++ b/core/decrypt/decrypt_linux.go
@@ -6,6 +6,7 @@ import (
 	"crypto/sha256"
 	"encoding/asn1"
 	"encoding/hex"
+	"errors"
 
 	"hack-browser-data/log"
 
@@ -100,21 +101,30 @@ func DecodeMeta(decodeItem []byte) (pbe MetaPBE, err error) {
 	return
 }
 
-func DecodeNss(nssA11Bytes []byte) (pbe NssPBE, err error) {
+func DecodeNss(nssA11Bytes []byte) (nss interface{}, err error) {
+	var pbe NssPBE
 	_, err = asn1.Unmarshal(nssA11Bytes, &pbe)
+
 	if err != nil {
-		log.Error(err)
-		return
+		return DecodeMeta(nssA11Bytes)
 	}
-	return
+
+	return pbe, err
 }
 
 func Meta(globalSalt, masterPwd []byte, pbe MetaPBE) ([]byte, error) {
 	return decryptMeta(globalSalt, masterPwd, pbe.EntrySalt, pbe.Encrypted)
 }
 
-func Nss(globalSalt, masterPwd []byte, pbe NssPBE) ([]byte, error) {
-	return decryptNss(globalSalt, masterPwd, pbe.IV, pbe.EntrySalt, pbe.Encrypted, pbe.IterationCount, pbe.KeySize)
+func Nss(globalSalt, masterPwd []byte, pbe interface{}) ([]byte, error) {
+	switch pbe.(type) {
+	case NssPBE:
+		return decryptNss(globalSalt, masterPwd, pbe.(NssPBE).IV, pbe.(NssPBE).EntrySalt, pbe.(NssPBE).Encrypted, pbe.(NssPBE).IterationCount, pbe.(NssPBE).KeySize)
+	case MetaPBE:
+		return decryptMeta(globalSalt, masterPwd, pbe.(MetaPBE).EntrySalt, pbe.(MetaPBE).Encrypted)
+	default:
+		return nil, errors.New("decrypt nss failed")
+	}
 }
 
 func decryptMeta(globalSalt, masterPwd, entrySalt, encrypted []byte) ([]byte, error) {

